File: src/main/java/easy/StringIntegerIsBalanced.java
package easy;

public class StringIntegerIsBalanced {
  public boolean solve(String num) {
    int even = 0;
    int odd = 0;
    for (int i = 0; i < num.length(); i++) {
      if (i % 2 == 0) {
        even += Character.getNumericValue(num.charAt(i));
      } else {
        odd += Character.getNumericValue(num.charAt(i));
      }
    }
    return even == odd;
  }
}

File: src/main/php/Easy/StringIntegerIsBalanced.php
<?php

declare(strict_types=1);

namespace Src\Easy;

class StringIntegerIsBalanced
{
    public function transformStringIntegerIsBalanced(string $num): bool
    {
        $even = 0;
        $odd = 0;
        for ($i = 0; $i < strlen($num); $i++) {
            if ($i % 2 === 0) {
                $even += intval($num[$i]);
            } else {
                $odd += intval($num[$i]);
            }
        }

        return $even === $odd;
    }

    public function solve(string $num): bool
    {
        return $this->transformStringIntegerIsBalanced($num);
    }
}
